core/src/main/java/org/apache/accumulo/core/security/ColumnVisibility.java

```diff
@@ -505,6 +523,19 @@ public ColumnVisibility(byte[] expression) {
     validate(expression);
   }
 
+  /**
+   * Creates a new column visibility by performing a deep copy of an existing column visibility
+   * object
+   *
+   * @param visibility ColumnVisibility object
+   * @since 2.1.4
+   */
+  public ColumnVisibility(ColumnVisibility visibility) {
+    byte[] incomingExpression = visibility.expression;
+    this.expression = Arrays.copyOf(incomingExpression, incomingExpression.length);
+    this.node = new Node(visibility.node);
```

Comment:
I made a comment in #5293 that is applicable here also. I think it's possible that another thread could be mutating the parse tree (visibility.node) while this thread is trying to make a copy of the ColumnVisibility because the parse tree is accessible and mutable.

Reply:
I am experimenting w/ adding an immutable parse tree to accumulo-access as a possible way to address this.